queries.yaml

```diff
@@ -794,6 +794,29 @@ post_process:
       where: >-
         kind == 'lake'
 
+  # remove names per kind for a given zoom
+  # before generating any label placements
+  - fn: vectordatasource.transform.drop_properties
+    params:
+      source_layer: water
+      start_zoom: 0
+      end_zoom: 16
+      # short-hand for "name" property in the list below means all name-like
+      # properties.
+      all_name_variants: true
+      properties:
+        - name
+      where: >-
+        (kind == 'sea' and zoom < 4) or
+        (kind == 'lake' and zoom < 5) or
+        (kind == 'river' and zoom <  12) or
+        (kind == 'canal' and zoom <  13) or
+        (kind == 'dam' and zoom <  14) or
+        (kind == 'stream' and zoom <  14) or
+        (kind == 'ditch' and zoom <  15) or
+        (kind == 'drain' and zoom <  15) or
+        (properties is not None and properties.get('is_tunnel') is True)
```

Comment:
The full list of water kinds is:

> basin, bay, canal, ditch, dock, drain, fjord, fountain, lake, playa, reef, river, riverbank, strait, stream, swimming_pool, water

So let's expand this list to include those

```
        (kind == 'sea' and zoom < 4) or
        (kind == 'bay' and zoom < 5) or
        (kind == 'fjord' and zoom < 5) or
        (kind == 'strait' and zoom < 5) or
        (kind == 'lake' and zoom < 5) or
        (kind == 'playa' and zoom < 6) or
        (kind == 'reef' and zoom < 6) or
        (kind == 'river' and zoom <  12) or
        (kind == 'riverbank' and zoom <  12) or
        (kind == 'water' and zoom <  12) or
        (kind == 'canal' and zoom <  13) or
        (kind == 'basin' and zoom <  13) or
        (kind == 'dock' and zoom <  13) or
        (kind == 'dam' and zoom <  14) or
        (kind == 'stream' and zoom <  14) or
        (kind == 'ditch' and zoom <  15) or
        (kind == 'drain' and zoom <  15) or
        (kind == 'swimming_pool' and zoom <  15) or
        (kind == 'fountain' and zoom <  15) or
        (properties is not None and properties.get('is_tunnel') is True)
```

Reply:
This addresses low zoom `basin`, `dock` and `straight` points I still see around London and the English Channel during QA.

Reply:
Addressed via https://github.com/tilezen/vector-datasource/pull/2010/commits/77e663c183b3ff687215386ce9ad219a8301f4fd.